src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double SIZE_INCREASE_FACTOR = 1.5;
+    private Object[] elementData; // should we use transient keyword?
```

Comment:
| should we use transient keyword?
for what purpose?

Reply:
For serialization. I don't know for sure, but original ArrayList uses it. That's why I'm asking